pkg/web3/sdk_v2.go

```diff
@@ -0,0 +1,225 @@
+	if err != nil {
+		return nil, err
+	}
+
+	lilypadContractRegistry, err := lilypadcontractregistry.NewLilypadContractRegistry(common.HexToAddress(options.LilypadContractRegistryAddress), client)
+	if err != nil {
+		return nil, err
+	}
+
+	lilypadStorage, err := lilypadstorage.NewLilypadStorage(common.HexToAddress(options.LilypadStorageAddress), client)
+	if err != nil {
+		return nil, err
+	}
+
+	lilypadUser, err := lilypaduser.NewLilypadUser(common.HexToAddress(options.LilypadUserAddress), client)
+	if err != nil {
+		return nil, err
+	}
+
+	lilypadTokenomics, err := lilypadtokenomics.NewLilypadTokenomics(common.HexToAddress(options.LilypadTokenomicsAddress), client)
+	if err != nil {
+		return nil, err
+	}
+
+	lilypadToken, err := lilypadtoken.NewLilypadToken(common.HexToAddress(options.LilypadL2TokenAddress), client)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ContractsV2{
+		LilypadPaymentEngine:    lilypadPaymentEngine,
+		LilypadProxy:            lilypadProxy,
+		LilypadContractRegistry: lilypadContractRegistry,
+		LilypadModuleDirectory:  lilypadModuleDirectory,
+		LilypadStorage:          lilypadStorage,
+		LilypadToken:            lilypadToken,
+		LilypadUser:             lilypadUser,
+		LilypadTokenomics:       lilypadTokenomics,
+	}, nil
+}
+
+func NewContractSDKV2(ctx context.Context, options Web3Options, tracer trace.Tracer) (*Web3SDKV2, error) {
+	displayOpts := options
+	displayOpts.PrivateKey = "*********"
+	logger.Debug().Msgf("NewContractSDK: %+v", displayOpts)
+
+	client, err := getEthClient(ctx, options, tracer)
+	if err != nil {
+		return nil, err
+	}
+
+	privateKey, err := ParsePrivateKey(options.PrivateKey)
+	if err != nil {
+		return nil, err
+	}
+
+	callOpts := &bind.CallOpts{
+		Pending:     false,
+		From:        common.Address{},
+		BlockNumber: nil,
+		Context:     nil,
+	}
+
+	transactOpts, err := bind.NewKeyedTransactorWithChainID(privateKey, big.NewInt(int64(options.ChainID)))
+	if err != nil {
+		return nil, err
+	}
+	contracts, err := NewContractsV2(options, client, callOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	web3SDK := &Web3SDKV2{
+		PrivateKey:   privateKey,
+		Options:      options,
+		Client:       client,
+		CallOpts:     callOpts,
+		TransactOpts: transactOpts,
+		Contracts:    contracts,
+		Log:          logger,
+	}
+
+	return web3SDK, nil
+}
+
+func (sdk *Web3SDKV2) getEthClient(ctx context.Context, options Web3Options, tracer trace.Tracer) (*ethclient.Client, error) {
+	ctx, span := tracer.Start(ctx, "get_ethclient", trace.WithAttributes(attribute.Int("web3.chain_id", options.ChainID)))
+	defer span.End()
+
+	rpcs := strings.Split(options.RpcURL, ",")
+	var err error
+	var parsedURL *url.URL
+	var client *ethclient.Client
+	for _, u := range rpcs {
+		parsedURL, err = url.Parse(u)
+		if err != nil {
+			logger.Warn().Msgf("Unable to parse web3 RPC URL: %v", err)
```

Comment:
```suggestion
			logger.Warn().Err(err).Msg("Unable to parse web3 RPC URL")
```
Could we log the error with an `Err` call to add it as a key-value pair?